tests/test_on_data/test_gin_ecephys.py

```diff
@@ -164,7 +170,8 @@ class TestConverter(NWBConverter):
 
     @parameterized.expand(input=parameterized_recording_list, name_func=custom_name_func)
     def test_convert_recording_extractor_to_nwb(self, data_interface, interface_kwargs):
-        nwbfile_path = str(self.savedir / f"{data_interface.__name__}.nwb")
+        x = np.random.randint(10000)
```

Comment:
So, I was having problems with the test because both neuroscope tests (differentiated by backend) have the same nwb_file name. Do you a better way? I have the feeling that there is something with the name_function above that we could adapt but maybe you remember it better? @CodyCBakerPhD

Reply:
Yeah, just add `_se_backend_{interface_kwargs.get("spikeextractors_backend", False)}` to the f-string for the filename.

Aside from that, what's the problem? The fact that `overwrite=True` isn't used?

Reply:
The file gets locked for some reason. I was wondering if the tests are run in parallel.

Reply:
File locking would indicate a potentially bad context exit, should look into that.

> The files for both backends have the same name for their respective nwbfile

I know, but tests are usually run in sequence to `overwrite=True` generally takes care of it unless there are issues with closing the file.

Reply:
OK.
Where or how should I look at this? Any tips? (bad context exit)

Reply:
> Where or how should I look at this? Any tips? (bad context exit)

That can be tricky, all you can do is try to reproduce and isolate the exact line of the issue then step back and examine the most immediate calls that opened/closed the file, which appears like it may be the `get_traces` calls from the previous `NwbRecordingExtractor` opened on the file for round-trip tests, but I don't see anything in the code for that that would cause an issue.

I'd say for now, just name them uniquely to bypass the issue and then see if you can systematically and reliably reproduce the error, then raise a formal issue on it and we can investigate it separately.